crates/test-framework/src/telemetry/mod.rs

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::sync::{Arc, LazyLock};
+
+use anyhow::Result;
+
+use opentelemetry::metrics::Meter;
+
+use opentelemetry_sdk::metrics::exporter::PushMetricExporter;
+use opentelemetry_sdk::metrics::reader::MetricReader;
+use opentelemetry_sdk::{
+    Resource,
+    metrics::{SdkMeterProvider, data::ResourceMetrics},
+};
+
+use telemetry::exporter::TelemetryExporterBuilder;
+pub use telemetry::meter::{METER_PROVIDER, METER_PROVIDER_ONCE};
+use telemetry::reader::InitialReader;
+
+const ENDPOINT_CONST: &str = "https://telemetry.spiceai.io";
+
+pub static ENDPOINT: LazyLock<Arc<str>> = LazyLock::new(|| {
+    std::env::var("SPICEAI_TELEMETRY_ENDPOINT")
+        .unwrap_or_else(|_| ENDPOINT_CONST.into())
+        .into()
+});
+
+pub static METER: LazyLock<Meter> = LazyLock::new(|| METER_PROVIDER.meter("benchmarks_telemetry"));
+
+pub struct Telemetry {
+    reader: InitialReader,
+    resource: Resource,
+    setup: bool,
+}
+
+impl Telemetry {
+    #[must_use]
+    pub fn new(resource: &Resource) -> Self {
+        let reader = InitialReader::default();
+
+        let provider = SdkMeterProvider::builder()
+            .with_resource(resource.clone())
+            .with_reader(reader.clone())
+            .build();
+
+        let setup = if METER_PROVIDER_ONCE.set(Arc::new(provider)).is_err() {
+            println!("Testoperator metrics are disabled");
+            false
+        } else {
+            true
+        };
+
+        Self {
+            reader,
+            resource: resource.clone(),
+            setup,
+        }
+    }
+
+    pub async fn read(&self, api_key: Option<&str>) -> Result<()> {
```

Comment:
```suggestion
    pub async fn emit(&self, api_key: Option<&str>) -> Result<()> {
```

Perhaps a little clearer?

Reply:
Also should the API key be part of the `Telemetry` struct? Seems weird that we need to specify it each time.